src/metabase/util/malli/fn.clj

```diff
@@ -312,16 +312,29 @@
 
 ;; ------------------------------ Skipping Namespace Enforcement in prod ------------------------------
 
-;; FIXME: is there a cleaner way to setup `namespaces-toskip`?
-;; I was thinking we could `swap!` it from the namespaces themselves with:
-;; `(swap namespaces-toskip conj *ns*)`
-(def namespaces-toskip
+(defn instrument-ns?
   "Used to track namespaces to not enforce malli schemas on with `mu.fn/fn`."
-  (atom #{"metabase.lib.metadata.calculation"}))
-
-(def ^:private ^:dynamic *skip-ns-decision-fn*
-  (core/fn skip-ns-decision-fn []
-    (and config/is-prod? (contains? @namespaces-toskip (str *ns*)))))
+  [namespace]
+  (let [lib-and-middleware [#"^metabase\.lib\..*"
+                            #"^metabase\.query-processor\.middleware\..*"]
+        matches?           (core/fn [namespace regexes]
+                             (let [n (-> namespace ns-name str)]
+                               (some #(re-matches % n) regexes)))
+        ad-hoc             #{'metabase.pulse.render}
+        m                  (meta namespace)]
+    (cond (:instrument/always m)                  true
+          (:instrument/never m)                   false
+          (matches? namespace lib-and-middleware) false
+          (contains? ad-hoc (ns-name namespace))  false
+          :else                                   true)))
+
+(def ^:dynamic *skip-ns-decision-fn*
+  (core/fn skip-ns-decision-fn [namespace]
+    (and #_config/is-prod? config/is-dev? ;;nocommit
```

Comment:
we need to remove the nocommit and switch from `is-dev?` to `is-prod?`

Reply:
done